client/api_bucket.go

```diff
@@ -761,3 +777,54 @@ func (c *client) CancelMigrateBucket(ctx context.Context, bucketName string, opt
 
 	return c.SubmitProposal(ctx, []sdk.Msg{cancelBucketMsg}, opts.ProposalDepositAmount, opts.ProposalTitle, opts.ProposalSummary, types.SubmitProposalOptions{Metadata: opts.ProposalMetaData, TxOption: opts.TxOpts})
 }
+
+// ListBucketsByPaymentAccount list bucket by payment account
+func (c *client) ListBucketsByPaymentAccount(ctx context.Context, paymentAccount string, opts types.ListBucketsByPaymentAccountOptions) (types.ListBucketsByPaymentAccountResult, error) {
+
+	params := url.Values{}
+	params.Set("payment-buckets", "")
+	params.Set("payment-account", paymentAccount)
+
+	reqMeta := requestMeta{
+		urlValues:     params,
+		contentSHA256: types.EmptyStringSHA256,
+	}
+
+	sendOpt := sendOptions{
+		method:           http.MethodGet,
+		disableCloseBody: true,
+	}
+
+	endpoint, err := c.getEndpointByOpt(&types.EndPointOptions{
+		Endpoint:  opts.Endpoint,
+		SPAddress: opts.SPAddress,
+	})
+	if err != nil {
+		log.Error().Msg(fmt.Sprintf("get endpoint by option failed %s", err.Error()))
+		return types.ListBucketsByPaymentAccountResult{}, err
+
+	}
+
+	resp, err := c.sendReq(ctx, reqMeta, &sendOpt, endpoint)
+	if err != nil {
+		return types.ListBucketsByPaymentAccountResult{}, err
+	}
+	defer utils.CloseResponse(resp)
+
+	buf := new(strings.Builder)
+	_, err = io.Copy(buf, resp.Body)
+	if err != nil {
+		log.Error().Msgf("the list bucket by payment account:%s failed: %s", paymentAccount, err.Error())
+		return types.ListBucketsByPaymentAccountResult{}, err
+	}
+
+	buckets := types.ListBucketsByPaymentAccountResult{}
+	bufStr := buf.String()
+	err = xml.Unmarshal([]byte(bufStr), &buckets)
+	if err != nil {
+		log.Error().Msgf("the list bucket by payment account:%s failed: %s", paymentAccount, err.Error())
+		return types.ListBucketsByPaymentAccountResult{}, err
```

Comment:
remove the log , return types.ListBucketsByPaymentAccountResult{}, errors.New("unmarshal response error"+ err.Error())

Reply:
fixed